canteen/canteen-bootstrap/src/main/go/main.go

```diff
@@ -22,7 +23,19 @@ func main() {
 
 	binary, err := exec.LookPath("java")
 	if err != nil {
-		log.Fatalf("Java execution error: %v", err)
+		javaHome, javaHomeSet := os.LookupEnv("JAVA_HOME")
+		if !javaHomeSet {
+			log.Fatalf("Java not found in PATH and JAVA_HOME not set: %v", err)
+		}
+
+		stat, err := os.Stat(javaHome)
+		if err != nil {
+			log.Fatalf("Java not found in JAVA_HOME: %v", err)
+		} else if !stat.IsDir() {
+			log.Fatalf("JAVA_HOME is not a directory: %v", err)
+		} else {
+			binary = path.Join(javaHome, "bin", "java")
```

Comment:
Should we extend the checks here to whether `$JAVA_HOME/bin/java` exists, is a regular file, and is executable or is this good enough?

Reply:
I think this is good enough. We are going to try to execute it further down, which will fail if `$JAVA_HOME/bin/java` is missing, is a directory, or is not executable.